### PR TITLE
kanidm: fix version

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Flakebi — CHANGES_REQUESTED** [review]
~~This is **not an unstable version**. It’s from a release branch.
I’m fine with adding the date, but the `unstable` here is wrong.~~

scratch that, it seems like nixpkgs wants to consider it unstable nonetheless.

**Comment by adamcstephens** [comment]
I don’t feel like the guidelines properly address this situation. This is not unstable, this is not an arbitrary commit chosen by us to bump to for some specific reason. These commits fix real issues identified by upstream. 

Upstream considers this the same release, and their current flow backports select commits into a branch for the release. I’ve already had multiple discussions with them about properly tagging these as releases, and my understanding is that this will happen in future releases. Unfortunately we’re not in the future yet, and needed to pull in changes that fix upgrades to this version for current users. 

If we are going to change the version then I’d much prefer @yu-re-ka’s suggestion as again, unstable is the wrong label. We may also need to update the guidelines, but I refuse to accept that there is no flexibility.

**Review by adamcstephens — CHANGES_REQUESTED** [review]
See comment, unstable is the wrong label to apply here.

File: pkgs/servers/kanidm/default.nix [hunk 1]
```diff
@@ -19,7 +19,7 @@ let
 in
 rustPlatform.buildRustPackage rec {
   pname = "kanidm";
-  version = "1.1.0-rc.16";
+  version = "1.1.0-rc.16-unstable-2024-03-02";
```
Comment (SuperSandro2000) [inline pkgs/servers/kanidm/default.nix:22]:
```suggestion
  version = "1.1.0-rc.16-5-g4c88d6b27";
```

Comment (adamcstephens) [reply pkgs/servers/kanidm/default.nix:22]: I’m not sure what git hash that is, but this package has already moved again. https://github.com/NixOS/nixpkgs/pull/299080

**Comment by SuperSandro2000** [comment]
Well, I just wanted to demonstrate what would be the correct value to put here, I really don't have the time to chase after this.

**Pull request closed without merging**